crates/bevy_sprite/src/tilemap_chunk/tilemap_chunk_material.wgsl

```diff
@@ -0,0 +1,58 @@
+#import bevy_sprite::{
+    mesh2d_functions as mesh_functions,
+    mesh2d_view_bindings::view,
+}
+
+struct Vertex {
+    @builtin(instance_index) instance_index: u32,
+    @location(0) position: vec3<f32>,
+    @location(1) uv: vec2<f32>,
+    @location(5) tile_index: u32,
+};
+
+struct VertexOutput {
+    @builtin(position) position: vec4<f32>,
+    @location(0) uv: vec2<f32>,
+    @location(1) tile_index: u32,
+}
+
+@group(2) @binding(0) var tileset: texture_2d_array<f32>;
+@group(2) @binding(1) var tileset_sampler: sampler;
+@group(2) @binding(2) var tile_indices: texture_2d<u32>;
+
+@vertex
+fn vertex(vertex: Vertex) -> VertexOutput {
+    var out: VertexOutput;
+    
+    let world_from_local = mesh_functions::get_world_from_local(vertex.instance_index);
+    let world_position = mesh_functions::mesh2d_position_local_to_world(
+        world_from_local,
+        vec4<f32>(vertex.position, 1.0)
+    );
+
+    out.position = mesh_functions::mesh2d_position_world_to_clip(world_position);
+    out.uv = vertex.uv;
+    out.tile_index = vertex.tile_index;
+
+    return out;
+}
+
+@fragment
+fn fragment(in: VertexOutput) -> @location(0) vec4<f32> {
+    let chunk_size = textureDimensions(tile_indices, 0);
+    let tile_xy = vec2<u32>(
+        in.tile_index % chunk_size.x,
```

Comment:
integer modulo typically expands to a buttload of instructions, if we know textureDimensions are always gonna be power of two we can optimize these, but the GPU can't

Reply:
I don't think there's a power of two constraint now is there? Would we want to impose that constraint for the performance gain? Or would we maybe want to expand the texture to the nearest power of two when preprocessing?